api/v1alpha1/catalogentry_types.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	kcpv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CatalogEntrySpec defines the desired state of CatalogEntry
+type CatalogEntrySpec struct {
+	// The workspace name where the APIExport locates
+	Workspace string `json:"workspace,omitempty"`
+	// The name of the APIExport
+	ExportName string `json:"exportName,omitempty"`
```

Comment:
these should be the APIBinding's export reference type

Reply:
This should be the `WorkspaceExportReference` as present in `APIBinding`. But the producer would be filing this with the APIExport's ws and name, whereas the APIBinding will reference to the CE details when it wants to bind to it. The operator will do the work of mapping the CE to the right APIExport. Is this understanding, right?

Reply:
Currently the ExportReference is "workspace+exportName"-only. But this will change. We will add the by-some-id binding variant. IMO this field here should be the blueprint for an APIBinding, hence reusing the ExportReference of the binding.

The operator/controller will not be involved in the actual binding process. That will be done by some CLI plugin. So the CLI plugin will then just copy the ExportReference over into a binding.

Reply:
My original thought was to use the direct fields under `ExportReference`/`WorkspaceExportReference` given they look like wrapper objects on top of those relevant fields at the moment. However, if we feel using `ExportReference` is more convenient, then I can certainly do so.

Reply:
> CatalogEntry:APIExport a 1:1 relation or rather 1:n ?

Maybe we should be flexible here and always use an array instead of a single value. Then we don't have to come back later and change from 1 to n.

> How will we represent dependencies?

This probably deserves a separate discussion, but I would imagine we'd want to allow CatalogEntry authors to specify one or both of the following:
- other catalog entries
- API group-resources